### PR TITLE
Fix various YAML-related bugs

### PR DESCRIPTION
**Pull request by dbendilas**
Problem and/or solution
-----------------------
There were 3 different scenarios in which the compilation of the
YAML front matter section was failing. All 3 had to do with wrapping
the string in quotes, and all were fixed.

- Wrap in double quotes if the string starts with `[` but does not end
  in `]`
- Wrap in double quotes if the string starts with `@`
- Wrap in single quotes if the string starts with double quotes but
  does not end in double quotes

How to test
-----------
Unit tests.

Reviewer checklist
------------------

Code:
* [ ] Change is covered by unit-tests
* [ ] Code is well documented, well styled and is following [best practices](https://tem.transifex.com)
* [ ] Performance issues have been taken under consideration
* [ ] Errors and other edge-cases are handled properly

PR:
* [ ] Problem and/or solution are well-explained
* [ ] Commits have been squashed so that each one has a clear purpose
* [ ] Commits have a proper commit message [according to TEM](https://tem.transifex.com/github-guide.html#working-on-a-feature)

### CODE REVIEW

File: openformats/formats/github_markdown_v2.py [hunk 1]
```diff
@@ -135,3 +112,129 @@ def parse(self, content, **kwargs):
...
+
+    def _should_wrap_in_quotes(self, string):
+        """Check if the given string should be wrapped in quotes.
+
+        In order to decide if wrapping is necessary, it takes into account
+        various parameters, such as what character the string starts and ends with,
+        whether or not it contains special characters, etc.
+
+        :param unicode string: the string to check
+        :return: a tuple that should if wrapping is needed and the character that
```
Comment (Diontsoumas) [inline openformats/formats/github_markdown_v2.py]: This return sentence seems incomplete. The tuple should what?

File: openformats/formats/github_markdown_v2.py [hunk 2]
```diff
@@ -135,3 +112,129 @@ def parse(self, content, **kwargs):
...
+        :param str quote_char: the character to use for wrapping
+        :return: the new string, wrapped in quotes if needed
+        :rtype: unicode
+        """
+        if quote_char == u'"':
+            string = string.replace(quote_char, self.BACKSLASH + quote_char)
+        elif quote_char == u"'":
+            string = string.replace(quote_char, quote_char * 2)
+        else:
+            raise ValueError(
```
Comment (Diontsoumas) [inline openformats/formats/github_markdown_v2.py]: Please add `:raise:` to the docstring

File: openformats/tests/formats/github_markdown_v2/test_github_markdown.py [hunk 3]
```diff
@@ -29,3 +31,104 @@ def test_parse(self):
         content_with_tab = self.handler.parse(content=u"# foo	bar")
         content_with_spaces = self.handler.parse(content=u"# foo    bar")
         self.assertEqual(content_with_tab[0], content_with_spaces[0])
+
+
+class GithubMarkdownV2CustomTestCase(unittest.TestCase):
+
```
Comment (dbendilas) [inline openformats/tests/formats/github_markdown_v2/test_github_markdown.py]: @dbendilas note to self: add a docstring here.

File: openformats/formats/github_markdown_v2.py (hunk 2 above)
Comment (dbendilas) [reply openformats/formats/github_markdown_v2.py]: Good catch!
